### PR TITLE
runtests using current ./lib and ./t directories

### PR DESCRIPTION
**Pull request by atoomic**
Add current ./lib and ./t directories to @INC when running tests
avoiding set_up_inc boilerplate in many tests.

### CODE REVIEW

**Comment by jkeenan** [comment]
I oppose the change suggested in this pull request.  It increases the minimum verbosity of commands in cases where I just want to run a test using the built `perl`.

At the present time, with this "boilerplate" in the test file, I can say this:
```
$ ./perl t/io/say.t
1..13
ok 1
ok 2
ok 3
ok 4
ok 5
ok 6
ok 7
ok 8
ok 9
ok 10
ok 11
ok 12
ok 13 - CORE::say without feature.pm
```
Which to me is DWIMming.

But I can't say that anymore if the `set_up_inc` line is eliminated.  In that case, I get:
```
$ ./perl t/io/say.t
Can't locate Errno.pm in @INC (you may need to install the Errno module) (@INC contains: /usr/local/lib/perl5/site_perl/5.31.10/amd64-freebsd-thread-multi /usr/local/lib/perl5/site_perl/5.31.10 /usr/local/lib/perl5/5.31.10/amd64-freebsd-thread-multi /usr/local/lib/perl5/5.31.10) at (eval 9) line 1.
BEGIN failed--compilation aborted at (eval 9) line 1.
BEGIN failed--compilation aborted at t/io/say.t line 9.
```
The workaround is not intuitive.  Because the test file changes into the `t/` subdirectory before executing the unit tests, we have to insert `lib` on the command-line.  But this will not suffice:
```
$ ./perl -Ilib t/io/say.t
Can't locate Errno.pm in @INC (you may need to install the Errno module) (@INC contains: lib /usr/local/lib/perl5/site_perl/5.31.10/amd64-freebsd-thread-multi /usr/local/lib/perl5/site_perl/5.31.10 /usr/local/lib/perl5/5.31.10/amd64-freebsd-thread-multi /usr/local/lib/perl5/5.31.10) at (eval 9) line 1.
BEGIN failed--compilation aborted at (eval 9) line 1.
BEGIN failed--compilation aborted at t/io/say.t line 9.
```
Instead, I have to include the relative path.
```
$ ./perl -I../lib t/io/say.t
1..13
ok 1
ok 2
ok 3
ok 4
ok 5
ok 6
ok 7
ok 8
ok 9
ok 10
ok 11
ok 12
ok 13 - CORE::say without feature.pm
```
In short, this p.r. makes simple stuff harder.  I strongly oppose it.

Thank you very much.
Jim Keenan

**Comment by atoomic** [comment]
Thanks @jkeenan for your feedback, I noticed that this would require the way people are manually running tests.

Here is one idea: right now we have a symlink for
` t/perl -> ../perl`
what about if `t/perl` was doing an `exec ../perl, -I., -I../lib, @ARGV`

note: alternatively I could also set an alias for it

so the existing one-liner you described would keep working and will avoid to add some incantation in most unit tests...

**Comment by atoomic** [comment]
tagged it as do not merge as we need more discussion on it 

note: alternatively `export PERL5LIB=$(PWD)/lib:$(PWD)/t` fixes it

**Comment by atoomic** [comment]
closing this PR as whatever is the solution adopted here, this will impact existing workflow 
tmtowtdi to run a Perl test :-) that would be nice if we could simply run every single test by using something like `./t/x/y.t`:-)
